test/integration/configuration.test.js

```diff
@@ -17,11 +17,24 @@ if (dialect === 'sqlite') {
 describe(Support.getTestDialectTeaser('Configuration'), () => {
   describe('Connections problems should fail with a nice message', () => {
     it('when we don\'t have the correct server details', () => {
-      const seq = new Sequelize(config[dialect].database, config[dialect].username, config[dialect].password, { storage: '/path/to/no/where/land', logging: false, host: '0.0.0.1', port: config[dialect].port, dialect });
       if (dialect === 'sqlite') {
+        const seq = new Sequelize(
+          config[dialect].database,
+          config[dialect].username,
+          config[dialect].password,
+          {
+            storage: '/path/to/no/where/land',
+            logging: false,
+            host: '0.0.0.1',
+            port: config[dialect].port,
+            dialect,
+            dialectOptions: { mode: sqlite3.OPEN_READONLY }
+          }
+        );
         // SQLite doesn't have a breakdown of error codes, so we are unable to discern between the different types of errors.
         return expect(seq.query('select 1 as hello')).to.eventually.be.rejectedWith(Sequelize.ConnectionError, 'SQLITE_CANTOPEN: unable to open database file');
       }
+      const seq = new Sequelize(config[dialect].database, config[dialect].username, config[dialect].password, { storage: '/path/to/no/where/land', logging: false, host: '0.0.0.1', port: config[dialect].port, dialect });
```

Comment:
Can you please refactor this part to avoid code duplication? Something like:

```js
const options = {
    logging: false,
    host: '0.0.0.1',
    port: config[dialect].port,
    dialect
};
const constructorArgs = [
    config[dialect].database,
    config[dialect].username,
    config[dialect].password,
    options
];

if (dialect === 'sqlite') {
    options.storage = '/path/to/no/where/land';
    options.dialectOptions = { mode: sqlite3.OPEN_READONLY };
    // ...
}
```

And you can call the constructor as `new Sequelize(...constructorArgs)`

It even has the advantage of not polluting the constructor with a `storage` option on the other dialects...